src/Components/SuperAdmins/Form/index.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from 'react';
+import styles from './form.module.css';
+import { Input } from '../../Shared/Inputs';
+import { Link, useParams, useLocation, useHistory } from 'react-router-dom';
+import Modal from '../../Shared/Modal';
+import Button from '../../Shared/Button';
+
+const Form = () => {
+  const { id } = useParams();
+  const [email, setEmail] = useState('');
+  const [pass, setPass] = useState('');
+  const [message, setMessage] = useState('');
+  const [showModal, setShowModal] = useState(false);
+  const history = useHistory();
+  const location = useLocation();
+
+  const getSuperAdmins = async (id) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins/${id}`);
+      const { data } = await response.json();
+      return data;
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  useEffect(() => {
+    if (location.pathname.includes('edit')) {
+      const setupForm = async () => {
+        const { email, password } = await getSuperAdmins(id);
+        setEmail(email);
+        setPass(password);
+      };
+      setupForm();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const onEdit = async (id) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins/${id}`, {
+        method: 'PUT',
+        body: JSON.stringify({ email, password: pass }),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const data = await response.json();
+      setMessage(data.message);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const onAdd = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins`, {
+        method: 'POST',
+        body: JSON.stringify({ email, password: pass }),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const data = await response.json();
+      setMessage(data.message);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    setShowModal(true);
+  };
+
+  const sendSuperAdmin = () => {
+    setShowModal(true);
+
+    if (location.pathname.includes('create')) {
+      onAdd({ email, pass });
+      setEmail('');
+      setPass('');
+      setShowModal(true);
+    }
+
+    if (location.pathname.includes('edit')) {
+      onEdit(id);
+    }
+  };
+  const reDirect = () => {
+    history.push('/super-admins');
+  };
+
+  return (
+    <div className={styles.formContainer}>
+      <Modal onClose={reDirect} isOpen={showModal} title={message} success />
```

Comment:
Try to make the redirect before the Modal appears if it was succesfull, if its not you should use another modal for the error, because if you fail in one of the inputs the modal appears and if you close it it redirects you to the table, this shouldn't be like that, if you have errors in the inputs you should stay on the form and fix them

Reply:
Ready. I left the onClose with the modal switch. And to the successful ones I do a redirect with a set time out